evmrpc/filter.go

```diff
@@ -349,6 +349,8 @@ func (f *LogFetcher) GetLogsByFilters(ctx context.Context, crit filters.FilterCr
 		defer wg.Done()
 		defer func() {
 			if r := recover(); r != nil {
+				mu.Lock()
```

Comment:
Is this being hit at all? I think recover() should never be hit in this logic, and it shouldn't be causing any dead lock issue